pkg/collect/all_logs.go

```diff
@@ -0,0 +1,136 @@
+package collect
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func AllLogs(c *Collector, logsCollector *troubleshootv1beta2.AllLogs) (CollectorResult, error) {
+	client, err := kubernetes.NewForConfig(c.ClientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	output := NewResult()
+
+	ctx := context.Background()
+	// If Namespaces list is empty, get all the namespaces
+	if len(logsCollector.Namespaces) == 0 || logsCollector.Namespaces[0] == "*" {
+		_, namespaceList, namespaceErrors := getAllNamespaces(ctx, client)
+		if len(namespaceErrors) > 0 {
+			output.SaveResult(c.BundlePath, getAllLogsErrorsFileName(logsCollector), marshalErrors(namespaceErrors))
+		}
+		logsCollector.Namespaces = getNamespaceNames(namespaceList)
+	}
+
+	for _, namespace := range logsCollector.Namespaces {
+		pods, podsErrors := listPodsInNamespace(ctx, client, namespace, logsCollector.Selector)
+		if len(podsErrors) > 0 {
+			output.SaveResult(c.BundlePath, getAllLogsErrorsFileName(logsCollector), marshalErrors(podsErrors))
+		}
+
+		if len(pods) > 0 {
```

Comment:
This `if` is unnecessary, it is enough to do `for` loop. If there are no items nothing will get executed.